src/App.tsx

```diff
@@ -1,7 +1,26 @@
 import './App.css';
+import List from './List';
+import { useState } from 'react';
 
 function App() {
-  return <div>App</div>;
+  const [page, change] = useState(0);
+  const [data] = useState([
+    { name: 'item1', hidden: true },
+    { name: 'item2', hidden: false },
+  ]);
+
+  return (
+    <div>
+      <List
+        page={page}
+        data={data}
+        setData={() => {
+          data.push({ name: 'item3', hidden: true });
+          change((p) => p + 1);
```

Comment:
Also this is kind of messed up. You give component a setData fn which secretly changes the page behind the scenes. I would separate these two